pkg/proxy/conntrack/cleanup.go

```diff
@@ -79,45 +97,53 @@ func CleanStaleEntries(ct Interface, ipFamily v1.IPFamily,
 			}
 		}
 
-		serviceIPEndpointIPs[svc.ClusterIP().String()] = endpointIPs
+		// a Service without endpoints does not require to clean the conntrack entries associated.
+		if endpointIPs.Len() == 0 {
+			continue
+		}
+
+		// we need to filter entries that are directed to a Service IP:Port frontend
+		// that does not have a backend as part of the endpoints IPs
+		portStr := strconv.Itoa(svc.Port())
```

Comment:
bugfix, Services are tuple of IP:Ports, when IP can be ClusterIP, LoadBalancerIP , ExternalIP and Ports are the ServicePorts.

For NodePort is different, any ip on the node (modulo the flag to choose the addresses) to a NodePort will be redirected to an endpoint